### PR TITLE
chore: Update response from `run` command

### PR DESCRIPTION
**Pull request by magreenbaum**
This handles incorrect deployment or schedule names in the command and returns appropriate message. 
```
Unable to find schedule with name 'dail' within a deployment named 'sandbox'
```

### CODE REVIEW

File: src/cloud-cli/meltano/cloud/api/client.py [hunk 1]
```diff
@@ -232,18 +232,28 @@ async def run_project(
...
+                    f"{self.config.internal_project_id}/"
+                    f"{deployment}/{job_or_schedule}"
+                )
+                await self._json_request(
+                    "POST",
+                    url,
+                )
+            except MeltanoCloudError as ex:
+                ex.response.reason = (
+                    f"Unable to find schedule with name {job_or_schedule!r} "
```
Comment (WillDaSilva) [inline src/cloud-cli/meltano/cloud/api/client.py:253]:
```suggestion
                    f"Unable to find schedule named {job_or_schedule!r} "
```

Nitpick - I know this error message is the same as the one from other parts of the Cloud CLI, but I've been meaning to update those too.

**Review by WillDaSilva — DISMISSED** [review]

File: src/cloud-cli/meltano/cloud/api/client.py [hunk 2]
```diff
@@ -232,18 +232,28 @@ async def run_project(
...
+                    f"{self.config.tenant_resource_key}/"
+                    f"{self.config.internal_project_id}/"
+                    f"{deployment}/{job_or_schedule}"
+                )
+                await self._json_request(
+                    "POST",
+                    url,
+                )
+            except MeltanoCloudError as ex:
+                ex.response.reason = (
```
Comment (edgarrmondragon) [inline src/cloud-cli/meltano/cloud/api/client.py:252]: Should we check the status code?

File: src/cloud-cli/meltano/cloud/api/client.py [hunk 3]
```diff
@@ -244,18 +244,30 @@ async def run_project(
...
+                    f"{self.config.tenant_resource_key}/"
+                    f"{self.config.internal_project_id}/"
+                    f"{deployment}/{job_or_schedule}"
+                )
+                await self._json_request(
+                    "POST",
+                    url,
+                )
+            except MeltanoCloudError as ex:
+                if ex.response.status == 400:  # noqa: WPS432
```
Comment (magreenbaum) [inline src/cloud-cli/meltano/cloud/api/client.py:264]: The response we're looking for from the runner would be a status code of `400`. Flakeheaven didn't like this though. Any issues with this @edgarrmondragon @WillDaSilva?

Comment (WillDaSilva) [inline src/cloud-cli/meltano/cloud/api/client.py:264]:
```suggestion
                if ex.response.status == HTTPStatus.BAD_REQUEST:
```

Comment (magreenbaum) [reply src/cloud-cli/meltano/cloud/api/client.py:264]: Oh I see. Thanks! 🤦‍♀️

Comment (magreenbaum) [reply src/cloud-cli/meltano/cloud/api/client.py:264]: Ah got it. Thanks!

**Review by WillDaSilva — APPROVED** [review]

File: src/cloud-cli/meltano/cloud/api/client.py [hunk 4]
```diff
@@ -244,18 +244,30 @@ async def run_project(
...
+                url = (
+                    "/run/v1/external/"
+                    f"{self.config.tenant_resource_key}/"
+                    f"{self.config.internal_project_id}/"
+                    f"{deployment}/{job_or_schedule}"
+                )
+                await self._json_request(
+                    "POST",
+                    url,
+                )
```
Comment (WillDaSilva) [inline src/cloud-cli/meltano/cloud/api/client.py:262]:
```suggestion
                await self._json_request("POST", url)
```

**Review by edgarrmondragon — APPROVED** [review]

**Pull request merged**